src/phasorpy/phasor.py

```diff
@@ -3070,13 +3092,157 @@ def _median(
     return numpy.nanmedian(real, **kwargs), numpy.nanmedian(imag, **kwargs)
 
 
-def _median_filter(
+def _2d_median_filter(
     real: ArrayLike,
     imag: ArrayLike,
+    /,
+    *,
     repeat: int = 1,
-    size: int | tuple[int] | None = 3,
+    size: int = 3,
+    axes: Sequence[int] | None = None,
+) -> tuple[NDArray[numpy.float64], NDArray[numpy.float64]]:
+    """Return the phasor coordinates after applying a 2D median filter.
+
+    Parameters
+    ----------
+    real : ndarray
+        Real components of the phasor coordinates.
+    imag : ndarray
+        Imaginary components of the phasor coordinates.
+    axes : int or sequence of int
+        Axes along which to apply the filter.
+    repeat : int, optional
+        Number of times to apply filter. The default is 1.
+    size : int or tuple of int, optional
+        The size of the median filter kernel. Default is 3.
+
+    Returns
+    -------
+    real : ndarray
+        Filtered real component of phasor coordinates.
+    imag : ndarray
+        Filtered imaginary component of phasor coordinates.
+
+    """
+    real = numpy.asarray(real, dtype=float).copy()
+    imag = numpy.asarray(imag, dtype=float).copy()
+
+    if axes is None:
+        axes = range(real.ndim)
+    if len(axes) != 2:
+        raise ValueError(
+            "Exactly 2 axes for filtering are required by this method."
+            f"Got {len(axes)} axes: {axes}"
+        )
```

Comment:
@cgohlke do you think that instead of raising this error it dispatches to `_median_filter_nd`? That way we also reduce the number of methods to maybe just `median` and `median_scipy`, in case we also want to keep scipy's?

Reply:
Yes.